autogen/oai/openai_utils.py

```diff
@@ -643,15 +649,22 @@ def config_list_from_dotenv(
     return config_list
 
 
-def retrieve_assistants_by_name(client: OpenAI, name: str) -> List[Assistant]:
-    """
-    Return the assistants with the given name from OAI assistant API
-    """
-    if ERROR:
-        raise ERROR
-    assistants = client.beta.assistants.list()
-    candidate_assistants = []
-    for assistant in assistants.data:
-        if assistant.name == name:
-            candidate_assistants.append(assistant)
-    return candidate_assistants
+try:
+    from openai import OpenAI
```

Comment:
openai>1.3 is required dependency: https://github.com/microsoft/autogen/blob/main/setup.py#L17

Reply:
Then should we remove all of the old code that depends on <1.1? @sonichi

Reply:
Addressed the immediate comment in this PR